src/coinhourbank/client.go

```diff
@@ -0,0 +1,193 @@
+package coinhourbank
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/coin"
+	"github.com/skycoin/skycoin/src/wallet"
+)
+
+// HourBankClient provides simplified access to the coin hour bank
+type HourBankClient struct {
+	bankURL string
+}
+
+// unspentOutput is UxOuts of transaction
+type unspentOutput struct {
+	Hash    string // Hash of unspent output
+	Address string // Address of receiver
+	Coins   uint64 // Number of coins
+	Hours   uint64 // Coin hours
+}
+
+// DepositHoursRequest models request for depositing coin hours
+type DepositHoursRequest struct {
+	UnspentOutputs []unspentOutput `json:"unspentOutput"`
+	CoinHours      uint64          `json:"coinHours"`
+}
+
+type coinjoinInput struct {
+	FromAddress string     // Address of transaction sender
+	UxOut       string     // UxOut HEX of transaction sender
+	Sign        cipher.Sig // Signature of input
+	InputIndex  int        // Index of this input in the transaction inputs array
+}
+
+// SignableTransaction is the entity for signing and publishing a transaction
+type SignableTransaction struct {
+	Transaction coin.Transaction `json:"transaction"`
+	Inputs      []coinjoinInput  `json:"inputs"`
+}
+
+// NewHourBankClient creates a new instance of HourBankClient
+func NewHourBankClient(bankURL string) *HourBankClient {
+	return &HourBankClient{
+		bankURL: bankURL,
+	}
+}
+
+// Balance returns balance for current account
+func (c *HourBankClient) Balance(account string) (uint64, error) {
```

Comment:
What is balance for? It can't check balance without auth, and there won't be auth from the desktop wallet

Reply:
oh I assumed its fine to know account balance without auth because it on the blockchain anyways

Reply:
Balance is really public unless you know all addresses the user deposits from.

Reply:
Cant edit for some reason, I meant *not really public.